src/java.desktop/windows/classes/sun/awt/windows/WFontConfiguration.java

```diff
@@ -249,8 +251,25 @@ private void initTables(String defaultEncoding) {
             textInputCharset = "HANGEUL_CHARSET";
         } else if ("x-windows-874".equals(defaultEncoding)) {
             textInputCharset = "THAI_CHARSET";
-        } else {
-            textInputCharset = "DEFAULT_CHARSET";
+        } else if (defaultEncoding.startsWith("UTF-8")) {
+            String lang = startupLocale.getLanguage();
+            String country = startupLocale.getCountry();
+            textInputCharset = switch(lang) {
+                case "ar" -> "ARABIC_CHARSET";
+                case "zh" -> {
+                    yield switch(country) {
+                        case "TW", "HK" -> "CHINESEBIG5_CHARSET";
+                        default -> "GB2312_CHARSET";
+                    };
+                }
+                case "ru" -> "RUSSIAN_CHARSET";
+                case "el" -> "GREEK_CHARSET";
+                case "iw" -> "HEBREW_CHARSET";
```

Comment:
It would be safe to specify both old(iw)/new(he) language codes for Hebrew, as "he" is now returned for Hebrew by default (https://bugs.openjdk.java.net/browse/JDK-8263202)

Reply:
ok  - fixed